magma/circuit.py

```diff
@@ -134,6 +136,37 @@ def __str__(cls):
             interface = f"({interface})"
         return f"{cls.__name__}{interface}"
 
+    def add_intermediate_value(cls, value, values):
```

Comment:
why are these cls functions? I don't see them modifying or introspecting `cls` at all

Reply:
Just for organization, I could make them helper functions outside the class, or make them staticmethods